tests/tasks/setup/kit/controlplane.yaml

```diff
@@ -0,0 +1,152 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: control-plane-setup
+  namespace: tekton-pipelines
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Kubernetes
+    tekton.dev/tags: CLI, kubectl
+    tekton.dev/displayName: "kubernetes actions"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: |
+    Setup a kubernetes control plane in the guest cluster.
+  params:
+  - name: cluster-name
+    default: "guest"
+    description: Name of the guest cluster
+  - name: kubernetes-version
+    default: "1.24"
+    description: Kubernetes version for the guest cluster 
+  - name: apiserver-replicas
+    default: "1"
+    description: Number of APIserver replicas
+  - name: apiserver-image
+    default: ""
+    description: Image of apiserver
+  - name: apiserver-parameters
+    default: ""
+    description: |
+      Parameters of the apiserver. e.g. '["--goaway-chance=0.01"]'
+  - name: apiserver-instance-type
+    default: "m5.2xlarge"
+    description: Instance type for the apiserver
+  - name: etcd-replicas
+    default: "3"
+    description: Number of ETCD replicas
+  - name: etcd-image
+    default: ""
+    description: Image of ETCD
+  - name: etcd-parameters
+    default: ""
+    description: Parameters of the ETCD container
+  - name: etcd-instance-type
+    default: "m5.xlarge"
+    description: Instance type for the ETCD
+  - name: cni-version
+    default: "v1.9"
+    description: Version of the CNI to be installed
+  - name: aws-ebs-csi-driver-version
+    default: release-1.13
+    description: The release version for aws ebs csi driver.
+  workspaces:
+  - name: config
+    mountPath: /config/
+  steps:
+  - name: setup-control-plane
+    image: bitnami/kubectl:1.24.5 # curl was removed in more recent versions
+    script: |
+      #!/bin/bash
+      echo "Approving CSRs"
+      kubectl certificate approve $(kubectl get csr | grep "Pending" | awk '{print $1}')  2>/dev/null || true
+      namespace=$(kubectl get ns $(params.cluster-name) -o yaml 2>/dev/null | grep phase | awk '{print $2}')
+      if [[ $namespace != "Active" ]]
+      then
+        echo "Create namespace"
+        kubectl create namespace $(params.cluster-name)
+      fi
+      echo "Setting up control plane"
+      cat <<EOF > /tmp/controlplane.yaml 
```

Comment:
let's take these files as params so it's much more dynamic than embedding it here like how we take it for service role tasks etc.